src/router.test.ts

```diff
@@ -509,6 +510,105 @@ describe('implementations', () => {
   });
 });
 
+describe('using middleware', () => {
+  test('type errors are caught when using middleware', () => {
```

Comment:
What is actually making the jest assertions here? Or are we just relying on the `@ts-expect-error` directives to act like a unit test?

Reply:
Yep, just the ts-expect-error directives. This pattern is used in many tests in this project.